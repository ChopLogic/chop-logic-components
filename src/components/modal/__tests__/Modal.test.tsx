import { describe, expect, it, vi } from 'vitest';
import { render, screen } from '@testing-library/react';
import ChopLogicModalLayout from '../elements/Layout';
import ChopLogicModalLayoutHeader from '../elements/Header';
import userEvent from '@testing-library/user-event';

describe('ChopLogicModal component', () => {
  const testLayoutProps = {
    id: 'test-id',
    title: 'Test modal',
    onClose: vi.fn(),
  };

  it('ChopLogicModalLayout should render correctly', () => {
    const { asFragment } = render(
      <ChopLogicModalLayout {...testLayoutProps} isOpened>
        <div>Test content</div>
      </ChopLogicModalLayout>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('ChopLogicModalLayout the dialog element', () => {
    render(
      <ChopLogicModalLayout {...testLayoutProps} isOpened>
        <div>Test content</div>
      </ChopLogicModalLayout>,
    );
    expect(screen.getByRole('dialog')).toBeInTheDocument();
  });

  it('ChopLogicModalLayoutHeader should display the modal title and the close button', () => {
    render(<ChopLogicModalLayoutHeader title={testLayoutProps.title} onClose={testLayoutProps.onClose} />);
    expect(screen.getByText(testLayoutProps.title)).toBeInTheDocument();
    expect(screen.getByRole('button')).toBeInTheDocument();
  });

  it('ChopLogicModalLayoutHeader should call onClose handler', async () => {
    render(<ChopLogicModalLayoutHeader title={testLayoutProps.title} onClose={testLayoutProps.onClose} />);
    await userEvent.click(screen.getByRole('button'));
    expect(testLayoutProps.onClose).toHaveBeenCalledOnce();
  });
});
