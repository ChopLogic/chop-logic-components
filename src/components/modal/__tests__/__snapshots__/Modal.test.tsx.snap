// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ChopLogicModal component > ChopLogicModalLayout should render correctly 1`] = `
<DocumentFragment>
  <div
    aria-modal="true"
    class="_window_cee13f"
    id="test-id"
    role="dialog"
  >
    <header
      class="_header_cee13f"
    >
      Test modal
      <button
        class="_button_48af00 _icon_48af00"
        type="button"
      >
        <span
          aria-hidden="true"
          class="chop-icon__close"
        />
        <span
          class="_text_48af00"
        />
      </button>
    </header>
    <div>
      <div>
        Test content
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ChopLogicModal component > should render correctly 1`] = `
<DocumentFragment>
  <div
    aria-modal="true"
    class="_window_cee13f"
    id="test-id"
    role="dialog"
  >
    <header
      class="modal-window__header"
      id="modal-window-heading"
    >
      Test modal
      <button>
        X
      </button>
    </header>
    <div
      aria-labelledby="modal-window-heading"
      role="region"
    >
      <div>
        Test content
      </div>
    </div>
  </div>
</DocumentFragment>
`;
